src/main/java/org/testeditor/fixture/web/json/BrowserSettingsManager.java

```diff
@@ -43,33 +46,112 @@
+     *          of JSON test file for browserSetup.json.
+     * @return 
+     *          A List of BrowserSetupElement
+     */
+    protected static List<BrowserSetupElement> getBrowserSettings(String fileName) {
+        
+        BrowserSetupReader reader = new BrowserSetupReader();
+        List<BrowserSetupElement> elements = reader.readElements(fileName);
+        
+        Platform currentPlattform = Platform.getCurrent();
+        checkRedundantElements(elements,currentPlattform);
+        return getBrowserSpecificSetting(elements, currentPlattform);
+    }
+    
+    protected static void checkRedundantElements(List<BrowserSetupElement> elements, Platform platform) {
+        List<BrowserSetting> duplicateSettings = new ArrayList<>();
+        
+        List<BrowserSetupElement> separatedBrowserSettingsForOs = separateBrowserSettingsForOs(elements, platform);
+        List<BrowserSetupElement> separateBrowserSettingsOsUnspecific = separateBrowserSettingsOsUnspecific(elements);
+        List<BrowserSetting> capabilitiesOsUnspecific = getBrowserSettingList(separateBrowserSettingsOsUnspecific,
+            CAPABILITIES); 
+        List<BrowserSetting> capabilitiesOsSpecific = getBrowserSettingList(separatedBrowserSettingsForOs,
+            CAPABILITIES); 
+        List<BrowserSetting> optionsOsUnspecific = getBrowserSettingList(separateBrowserSettingsOsUnspecific, 
+            OPTIONS);  
+        List<BrowserSetting> optionsOsSpecific = getBrowserSettingList(separatedBrowserSettingsForOs, 
+            OPTIONS);  
+        
+        // Look for duplicate capability entries and exit test when found. 
+        determineDuplicateEntries(duplicateSettings, capabilitiesOsUnspecific, capabilitiesOsSpecific);
+        // Look for duplicate option entries and exit test when found.
+        determineDuplicateEntries(duplicateSettings, optionsOsUnspecific, optionsOsSpecific);
+        
+        if (duplicateSettings.size() > 0) {
+            throw new RuntimeException("Duplicate entries existing in configuration file " 
+                + FILE_NAME + ". Entries are: " + duplicateSettings);
+        }
+    }
+    
+    protected static void determineDuplicateEntries(List<BrowserSetting> duplicateSettings, 
+            List<BrowserSetting> settingOsUnspecific, List<BrowserSetting> settingOsSpecific) {
+        for (BrowserSetting browserSetting : settingOsUnspecific) {
+            String capabilityName = browserSetting.getKey();
+            for (BrowserSetting browserSettingDuplicate : settingOsSpecific) {
+                String duplicateCapabilityName = browserSettingDuplicate.getKey();
+                if (capabilityName.equalsIgnoreCase(duplicateCapabilityName)) {
+                    duplicateSettings.add(browserSetting);
+                    duplicateSettings.add(browserSettingDuplicate);
+                } 
+            }
+        }
+    }
+    
+    protected static List<BrowserSetting> getBrowserSettingList(List<BrowserSetupElement> 
```

Comment:
why pass browserSettingsType? caller knows whether it is the one or the other and could call two separate functions